Core/Utilities/Profiler.py

```diff
@@ -0,0 +1,205 @@
+"""
+Profiling class for updated information on process status
+"""
+
+__RCSID__ = "$Id$"
+
+import psutil
+import datetime
+import errno
+
+from DIRAC import gLogger, S_OK, S_ERROR
+from DIRAC.Core.Utilities.DErrno import EEZOMBIE
+from DIRAC.Core.Utilities.DErrno import EENOPID
+from DIRAC.Core.Utilities.DErrno import EEEXCEPTION
+
+class Profiler( object ):
+  """
+  Class for profiling both general stats about a machine and individual processes.
+  Every instance of this class is associated to a single process by using its PID.
+  Calls to the different methods of the class will return the current state of the process.
+  """
+
+  def __init__( self, pid = None ):
+    """
+    :param str pid: PID of the process to be profiled
+    """
+    if pid:
+      try:
+        self.process = psutil.Process( int( pid ) )
+      except psutil.NoSuchProcess as e:
+        gLogger.error( 'No such process: %s' % e )
+    else:
+      self.process = None
+
+  def pid( self ):
+    """
+    Returns the process PID
+    """
+    if self.process:
+      return S_OK( self.process.pid )
+    else:
+      gLogger.error( 'No PID of process to profile' )
+      return S_ERROR( EENOPID, 'No PID of process to profile' )
+
+  def status( self ):
+    """
+    Returns the process status
+    """
+    if self.process:
+      try:
+        result = self.process.status()
+      except psutil.NoSuchProcess as e:
+        gLogger.error( 'No such process: %s' % e )
+        return S_ERROR( errno.ESRCH, 'No such process: %s' % e )
+      except psutil.ZombieProcess as e:
+        gLogger.error( 'Zombie process: %s' % e )
+        return S_ERROR( EEZOMBIE, 'Zombie process: %s' % e )
+      except psutil.AccessDenied as e:
+        gLogger.error( 'Access denied: %s' % e )
+        return S_ERROR( errno.EPERM, 'Access denied: %s' % e )
+      except Exception as e:
+        gLogger.error( e )
+        return S_ERROR( EEEXCEPTION, e )
+
+      return S_OK( result )
+    else:
+      gLogger.error( 'No PID of process to profile' )
+      return S_ERROR( EENOPID, 'No PID of process to profile' )
+
+  def runningTime( self ):
+    """
+    Returns the uptime of the process
+    """
+    if self.process:
+      try:
+        start = datetime.datetime.fromtimestamp( self.process.create_time() )
+        result = ( datetime.datetime.now() - start ).total_seconds()
+      except psutil.NoSuchProcess as e:
```

Comment:
I have more problem with non tested code. I am not going to spend time to make decorator. Feel free and do it